tools/autograd/templates/VariableType.h

```diff
@@ -42,14 +42,16 @@ using at::Quantizer;
 using ConstQuantizerPtr = const c10::intrusive_ptr<Quantizer>&;
 using c10::optional;
 
+// TODO: Once we have C++17, this can be a template function rather than a macro.
+#define UNPACK_TENSOR( t, pos ) \
```

Comment:
Do we really need a macro for this given that it is only used in the codegen anyways? You can replace the macro in the template with these two lines.

This used to be much more complex before the Tensor/Variable merge, but not much is left.

Reply:
I elected for the macro because I didn't want the manually defined functions to drift from the code-gen'd ones.

Reply:
They should be updated the same way. But it is know that whenever we change the codegen, the same update should be done on the manually defined functions.